File: clockinout.php
<?php

namespace Facebook\WebDriver;

use Facebook\WebDriver\Remote\DesiredCapabilities;
use Facebook\WebDriver\Remote\RemoteWebDriver;

require_once('../vendor/autoload.php');

// start firefox
$host = 'http://localhost:4444/wd/hub'; // this is the default
$desired_capabilities = DesiredCapabilities::firefox();
$desired_capabilities->setCapability('acceptSslCerts', false);
//TO RUN HEADLESS
$desired_capabilities->setCapability('moz:firefoxOptions', ['args' => ['-headless']]);

$driver = RemoteWebDriver::create($host, $desired_capabilities);

// navigate to 'https://clock.payrollservers.us/?wl=payday.payrollservers.us#/clock/web/login'
$driver->get('https://clock.payrollservers.us/?wl=payday.payrollservers.us#/clock/web/login');
// wait 10 sec
$driver->manage()->timeouts()->implicitlyWait(10);

// print the title of the current page
echo "The title is '" . $driver->getTitle() . "'\n";
echo "<br>";
// print the URI of the current page
echo "The current URI is '" . $driver->getCurrentURL() . "'\n";
echo "<br>";


//write 'tabeytanbgs' in the USERNAME box
$driver->findElement(WebDriverBy::id('Username'))
   ->sendKeys('username');

//write 'P@ssword1' in password box
$driver->findElement(WebDriverBy::id('Password'))
   ->sendKeys('password');

// click the link 'Login'
$Login = $driver->findElement(WebDriverBy::id('Login'));
$Login->click();
// wait 10 sec
$driver->manage()->timeouts()->implicitlyWait(10);

// clockin
//$clockin = "(//button[@id='ClockIn']/div)";
//$clockin = $driver->findElement(WebDriverBy::xpath($clockin));
//$clockin->click();

//clockout
//$clockout = "(//button[@id='ClockOut']/div)";
//$clockout = $driver->findElement(WebDriverBy::xpath($clockout));
//$clockout->click();

//signout
$signout = $driver->findElement(WebDriverBy::id('SignOut'));
$signout->click();

//close the browser
$driver->quit();


// print successfull clockin
echo "<br>";
echo "Successfull ClockIn at " . date("Y/m/d"). "<br>" . date("h:i:sa");
echo "<br>";

// print successfull clockout
echo "<br>";
echo "Successfull ClockOut at " . date("Y/m/d") . "<br>" . date("h:i:sa");
echo "<br>";


?>

File: clockinout.py
#!/bin/python

import datetime

from selenium import webdriver
from selenium.webdriver.common.desired_capabilities import DesiredCapabilities
from selenium.webdriver.firefox.options import Options
from selenium.webdriver.common.keys import Keys


options = Options()
options.headless = True
desired_caps = DesiredCapabilities.FIREFOX

driver = webdriver.Remote(
   command_executor='http://localhost:4444/wd/hub',
   desired_capabilities=desired_caps,
   options=options)

driver.get('https://clock.payrollservers.us/?wl=payday.payrollservers.us#/clock/web/login')
driver.implicitly_wait(10)

element = driver.find_element_by_id("Username")
element.clear()
element.send_keys("username")
element = driver.find_element_by_id("Password")
element.clear()
element.send_keys("password", Keys.ENTER)
driver.implicitly_wait(10)


#xpath1="//button[@id='ClockIn']"
#clockin = driver.find_element_by_xpath(xpath1)
#clockin.click()

#xpath2="//button[@id='ClockOut']"
#clockout = driver.find_element_by_xpath(xpath2)
#clockout.click()

signout = driver.find_element_by_id("SignOut")
signout.click()



driver.close()

currentDT=datetime.datetime.now()
print("\n\n\n\n\n\n\nClocked IN at " + str(currentDT))
print("\n\n\n\n\n\n\nClocked OUT at " + str(currentDT))  
